cpp/src/neighbors/sample_filter.cuh

```diff
@@ -140,4 +146,48 @@ void bitmap_filter<bitmap_t, index_t>::to_csr(raft::resources const& handle, csr
   raft::sparse::convert::bitmap_to_csr(handle, bitmap_view_, csr);
 }
 
+struct none_filter_args_t {};
+using bitset_filter_args_t =
+  std::tuple<const int64_t* const*, cuvs::core::bitset_view<uint32_t, int64_t>>;
+
+struct ivf_filter_dev {
+  filtering::FilterType tag_;
+
+  union ivf_filter_dev_args_variant {
+    none_filter_args_t none_filter_args;
+    bitset_filter_args_t bitset_filter_args;
+
+    _RAFT_HOST_DEVICE ivf_filter_dev_args_variant() {}
+    _RAFT_HOST_DEVICE ~ivf_filter_dev_args_variant() {}
+  } args_;
+
+  _RAFT_HOST_DEVICE ivf_filter_dev(none_filter_args_t args = {}) : tag_(FilterType::None)
+  {
+    new (&args_.none_filter_args) none_filter_args_t(args);
```

Comment:
Can't we just initialize using the `args_(args)` same as you initialize the `tag_` above (after you remove explicit constructor/destructor)?

Reply:
@achirkin @divyegala 
Changed to use the initializer list. We still need constructors in the union because `bitset_filter_args_t` is not trivially constructible and therefore does not have a default constructor. But we don't need the destructor anymore.